src/sentry/static/sentry/app/utils/apm.tsx

```diff
@@ -9,3 +16,61 @@ export function setTransactionName(name: string) {
     scope.setTag('ui.route', name);
   });
 }
+
+export async function normalizeTransactionName(
+  appRoutes: Router.PlainRoute[],
+  event: Sentry.Event
+): Promise<Sentry.Event> {
+  if (event.type !== 'transaction') {
+    return event;
+  }
+
+  // For JavaScript transactions, translate the transaction name if it exists and doesn't start with /
+  // using the app's react-router routes. If the transaction name doesn't exist, use the window.location.pathname
+  // as the fallback.
+
+  let prevTransactionName = event.transaction;
+
+  if (typeof prevTransactionName === 'string') {
+    if (prevTransactionName.startsWith('/')) {
+      return event;
+    }
+
+    set(event, ['tags', 'transaction.rename.source'], 'existing transaction name');
```

Comment:
Do we want this before https://github.com/getsentry/sentry/pull/18822/files#diff-f8570bef96092fcfcfc16de1804f940dR36?

Reply:
@billyvg nah I don't think so. I think, semantically, tags that begin `transaction.rename` should only apply to events that have undergone a transaction rename process.

Reply:
To clarify, we shouldn't be renaming transactions that begin with `/`. Since these transactions are what we'd expect. They've already been renamed at the React app level from within the lifecycle method.

Reply:
That makes sense, the word "existing" kind of threw me off (I was thinking existing mean it has a correct transaction name).